### ISSUE TITLE
More intuitive error mapping in python-flake8

### ISSUE THREAD

**Issue by erikkaplun** [flycheck/flycheck#241]
Currently, `E*` from `pep8` (via `flake8`) is reported as an error, while `F*` from `pyflakes` (via `flake8`) is reported as warning. This however makes little sense as `pep8` errors are, from an "IDE" point of view, simply style warnings, while `F*` from `pyflakes` is stuff like "undefined name" or "unused variable" which is mostly more or less a coding error not a warning.

For me, the following errors/warnings definition for `python-flake8` in `flycheck.el` solves the problem:

```
((error line-start
        (file-name) ":" line ":" (optional column ":") " "
         (message "F"                  ; Pyflakes in Flake8 >= 2.0
                 (one-or-more digit) (zero-or-more not-newline))
        line-end)
 (warning line-start
          (file-name) ":" line ":" (optional column ":") " "
          (message (or "E"            ; pep8 in Flake8 >= ???
                       "W"            ; Pyflakes in Flake8 < 2.0
                       "C"            ; McCabe in Flake >= 2.0
                       "N")           ; pep8-naming in Flake8 >= 2.0
                   (one-or-more digit) (zero-or-more not-newline))
          line-end)
```
